### ISSUE TITLE
form.match doesn't use coordinates from given schema

### ISSUE THREAD

**Issue by fsad** [creatale/node-fv#1]
Hi,

first let me thank you for this great library! It saved me a lot of time!

I am not sure what I am doing wrong or if it is not supported: 

I defined my schema as follows: 
```javascript
formSchema = {
    page: {
        x:0,
        y:0,
        width: 1833,
        height: 971
    },
    words: [],
    fields: [
        {
            path: 'age_1',
            type: 'text',
            box: {
                x: 385,
                y: 213,
                width: 65,
                height: 94
            }
        },
        {
            path: 'age_2',
            type: 'text',
            box: {
                x: 454,
                y: 213,
                width: 65,
                height: 94
            }
        },
            fieldValidator: function(value) {return true;}
        }
    ]
};
```
Whenever I call form.match like follows ... 

```javascript
form.match(formSchema, (function (_this) {
    return function (err, formData) {
        console.log(formData);
        //console.log(form.toObject());
        //return fs.writeFile('example.log.png', form.toImage().toBuffer('png'));
    };
})(this)); 
```
.. it doesn't match the given positions but has other x and y coordinates in the result set:

```javascript
age_1: 
   { value: 'i',
     confidence: 75,
     box: { x: 306, y: 297, width: 203, height: 96 },
     conflicts: [] },
  age_2: 
   { value: 'Jahre',
     confidence: 88,
     box: { x: 542, y: 234, width: 81, height: 25 },
     conflicts: [] } }
```

I double checked the positions/coordinates with photoshop - they are correct as set up in the schema.
Any idea?

Thanks a lot! 
Florian

**Comment by WolfgangFellger** [comment]
Sorry for the late reply. I'd need a complete example (input, schema, and cli command line) to comment on this properly; right now I do not quite get what you expect FV to do. Assuming that the attached image is your input, a few notes:

- There are currently a lot of hardcoded assumptions about dpi in the library. Your input should be in the 250-400dpi range for everything to work properly.
- Your sample image has a different format than your schema.
- Positions in the schema are merely taken as hints. There is always a bit of fuzzy matching, because at least our forms aren't printed or even fed into the scanner with millimeter precision ;)
- Tesseract will always be run globally, not just within specified fields (see below); these solid boxes *will* confuse it. If possible, you should design your form so you can filter these out before image processing (we use red ink, thus the removeRed filter).
- Most importantly: If you do not provide any fieldValidators, the matching process has very little to 'grip' on. The main feature of FV is that it takes into account the relative positions of fields in the schema to find even rather severe offsets in the printed form, but this requires a couple of unique matches (form text, or better, uniquely formatted fields). The fieldValidators provide the hints to do this.

In short, if you expected a pixel-accurate matching, this is currently not supported.

**Comment by kcollignon** [comment]
@WolfgangFellger just reading up on your comments above. Can you help clarify the best approach to take when the form does not have red ink, but there are multiple boxes? (think an invoice layout). What is the best use case for passing a schema? I was under the impression it would return the results back from what it matched in the schema from the X-Y coordinates. Thanks!

**Comment by WolfgangFellger** [comment]
Well, you have to get rid of these boxes using image preprocessing, or Tesseract will be confused. fv does not come with such a filter at the moment, but you can probably construct something out of dv.

Additionally, if all you want is running text recognization within hardcoded rectangles, there is little use to fv; you can simply crop and recognize them using `dv` directly. fv is really for fuzzy form matching.
